src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -38,85 +38,97 @@ public static string GetExpressionText(LambdaExpression expression, ExpressionTe
             }
 
             var containsIndexers = false;
-            // Split apart the expression string for property/field accessors to create its name
-            var nameParts = new Stack<string>();
             var part = expression.Body;
+            expressionText = string.Empty;
 
-            while (part != null)
+            if (part != null &&
+                (part.NodeType == ExpressionType.Call ||
+                part.NodeType == ExpressionType.ArrayIndex ||
+                part.NodeType == ExpressionType.MemberAccess))
             {
-                if (part.NodeType == ExpressionType.Call)
+                // Split apart the expression string for property/field accessors to create its name
+                var builder = new StringBuilder();
+                var parameters = expression.Parameters.ToArray();
```

Comment:
Only create this array when necessary. Probably best to do this in `GetIndexerInvocation()`.